File: LeetCode/Easy/0066-plus-one/0066-plus-one.go
func plusOne(digits []int) []int {
	c := 1
	n := len(digits)
	i := n - 1
	for i >= 0 {
		s := (digits[i] + c)
		digits[i] = (s) % 10
		c = (s) / 10
		if c == 0 {
			break
		}
		i--
	}
	ans := []int{}
	if c > 0 {
		ans = append(ans, c)
	}
	ans = append(ans, digits...)
	return ans
}

File: LeetCode/Easy/0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        reverse(digits.begin(), digits.end());
        int c = 1;
        int i = 0, sum = 0;
        while (i < digits.size()) {
            sum = digits[i] + c;
            // cout<<sum<<endl;
            digits[i] = (sum) % 10;
            c = (sum) / 10;
            i++;
        }
        if (c > 0) {
            digits.push_back(c);
        }

        reverse(digits.begin(), digits.end());
        return digits;
    }
};